torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -113,18 +126,23 @@ class AttributePropagator {
       LowerSimpleTuples(subgraph);
     };
 
+    std::unordered_map<std::string, std::unordered_set<std::string>>
+        interfacesToRetype;
+
     for (auto function : preservedMethods_) {
       GRAPH_DEBUG("Analyzing function: " + function->name());
       auto graph = toGraphFunction(*function).graph();
       optimizeSubGraphs(graph, applyInline);
       if (freezeInterfaces_) {
-        inlineInterfaceCalls(graph);
+        inlineInterfaceCalls(graph, interfacesToRetype);
       }
       // Record Attributes that are explicitly set in the module.
       // They cannot be folded.
       recordMutableAttrs(graph);
```

Comment:
@eellison suppose we put reassignInterfaceTypes immediately after inlineInterfaceCalls. Then we can have the following situation:

* preservedMethod1 does getattr(self, some_interface_module). It registers it as an interface to reassign type
* we reassign type for some_interface_module, so now the module object labels it as an interface type
* preservedMethod2 does setattr(self, some_interface_module). But when it looks it up in the module, it sees that some_interface_module is _not_ an interface (because the type was reassigned already). So it doesn't error out.

Is there anything that can go wrong with reassigning interface types after recordMutableAttrs?

Reply:
Yea, good point. Sorry the important part is that the calls themselves are inlined (and mutation is observed) not that the actual types on the modules are re-assigned.

Reply:
Actually just to be safe I reordered it anyway, just added an extra loop to do the recordMutableAttrs afterwards.